lib/puppet/provider/esx_firewall_ruleset/default.rb

```diff
@@ -0,0 +1,101 @@
+provider_path = Pathname.new(__FILE__).parent.parent
+require File.join(provider_path, 'vcenter')
+require 'rbvmomi'
+
+Puppet::Type.type(:esx_firewall_ruleset).provide(
+  :default, 
+  :parent => Puppet::Provider::Vcenter
+) do
+
+
+  @doc = "Manages ESXi firewall rulesets"
+
+  def enable
+    firewall_config.EnableRuleset(:id => @resource[:name])
```

Comment:
Thanks for the comment. I am not used to seeing the retrieve method so I didn't follow the logic trail properly.